src/Locator.php

```diff
@@ -226,9 +236,15 @@ protected function search_elements_by_tag($name, &$done, $feeds)
                     $headers = [
                         'Accept' => 'application/atom+xml, application/rss+xml, application/rdf+xml;q=0.9, application/xml;q=0.8, text/xml;q=0.8, text/html;q=0.7, unknown/unknown;q=0.1, application/unknown;q=0.1, */*;q=0.1',
                     ];
-                    $feed = $this->registry->create(File::class, [$href, $this->timeout, 5, $headers, $this->useragent, $this->force_fsockopen, $this->curl_options]);
-                    if ($feed->success && ($feed->method & \SimplePie\SimplePie::FILE_SOURCE_REMOTE === 0 || ($feed->status_code === 200 || $feed->status_code > 206 && $feed->status_code < 300)) && $this->is_feed($feed, true)) {
-                        $feeds[$href] = $feed;
+
+                    try {
+                        $file = $this->get_http_client()->request(Client::METHOD_GET, $href, $headers);
```

Comment:
How do we deal with redirects?

Reply:
The `File` class handles redirects in the constructor.